yal/optout.py

```diff
@@ -0,0 +1,289 @@
+                "delete saved": "state_delete_saved",
+                "skip": MainMenuApplication.START_STATE,
+            },
+            error=self._(GENERIC_ERROR),
+        )
+
+    async def state_stop_messages(self):
+        self.save_answer("state_optout_messages", True)
+        msg = self._(
+            "\n".join(
+                [
+                    "✅ B-Wise by Young Africa Live will stop sending you messages.",
+                    "------",
+                    "",
+                    "👩🏾 I'm sorry to see you go. It's been a pleasure talking to you.",
+                ]
+            )
+        )
+        await self.worker.publish_message(self.inbound.reply(msg))
+        return await self.go_to_state("state_optout_survey")
+
+    async def state_stop_notifications(self):
+        self.save_answer("state_optout_notifications", True)
+        msg = self._(
+            "\n".join(
+                [
+                    "✅ B-Wise by Young Africa Live will stop sending you messages.",
+                ]
+            )
+        )
+        self.worker.publish_message(self.inbound.reply(msg))
+        await asyncio.sleep(0.5)
+        return await self.go_to_state("state_optout_survey")
+
+    async def state_optout_survey(self):
+        question = self._(
+            "\n".join(
+                [
+                    "🛑 STOP MESSAGING ME",
+                    "*What can we do better?*",
+                    "------",
+                    "",
+                    "*👩🏾 We are always trying to improve.*",
+                    "*Could you tell us why you want to stop getting these messages?*",
+                    "",
+                    "Your answer will help us make this service better.",
+                    "",
+                    "1 - Getting too many messages",
+                    "2 - Find the service difficult to use",
+                    "3 - The messages are too long",
+                    "4 - The messages are boring/unoriginal/repetitive",
+                    "5 - The content is not relevant to me",
+                    "6 - Other",
+                    "7 - Rather not say",
+                ]
+            )
+        )
+        return WhatsAppListState(
+            self,
+            question=question,
+            button="Opt Out",
+            choices=[
+                Choice("message volume", self._("Too many messages")),
+                Choice("user-friendliness", self._("Difficult to use")),
+                Choice("irrelevant", self._("Content is irrelevant")),
+                Choice("boring", self._("Content is boring/repetitive")),
+                Choice("lengthy", self._("Messages are too long")),
+                Choice("other", self._("Other")),
+                Choice("none", self._("Rather not say")),
+                Choice("skip", self._("Skip")),
+            ],
+            next={
+                "message volume": "state_farewell_optout",
```

Comment:
We can handle this better using a function instead of a dict, example here: https://github.com/praekeltfoundation/vaccine-eligibility/blob/main/yal/onboarding.py#L438-L441